install_config/storage_examples/storage_classes_dynamic_provisioning.adoc

```diff
@@ -0,0 +1,322 @@
+.StorageClass Fast Spec Definition
+====
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: fast
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-ssd
+  zone: us-east1-d
+----
+====
+
+As a cluster-admin or storage-admin save both definitions to .yaml files, i.e. slow-gce.yaml and fast-gce.yaml and then create the StorageClasses.
+
+====
+----
+
+# oc create -f slow-gce.yaml 
+storageclass "slow" created
+
+# oc create -f fast-gce.yaml 
+storageclass "fast" created
+
+# oc get storageclass
+NAME       TYPE
+fast       kubernetes.io/gce-pd   
+slow       kubernetes.io/gce-pd   
+
+----
+====
+
+[IMPORTANT]
+====
+cluster-admin or storage-admins are responsible for relaying the correct _StorageClass_ name to the correct users/groups and projects
+====
+
+
+As a user, create a new project and claim the _StorageClass_
+First create the claim yaml definition, save it to a file (i.e. pvc-fast.yaml) and then execute it using _oc create_ command and finally
+check to see if your claim was bound.
+
+====
+----
+
+# oc new-project rh-eng
+
+----
+====
+
+.Claim for Fast Storage Spec Definition
+====
+[source,yaml]
+----
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+ name: pvc-engineering
+ annotations:
+   volume.beta.kubernetes.io/storage-class: fast
+spec:
+ accessModes:
+  - ReadWriteMany
+ resources:
+   requests:
+     storage: 10Gi
+----
+====
+
+====
+----
+
+# oc create -f pvc-fast.yaml 
+persistentvolumeclaim "pvc-engineering" created
+
+----
+====
+
+====
+----
+
+# oc get pvc
+NAME              STATUS    VOLUME                                     CAPACITY   ACCESSMODES   AGE
+pvc-engineering   Bound     pvc-e9b4fef7-8bf7-11e6-9962-42010af00004   10Gi       RWX           2m
+
+----
+====
+
+[IMPORTANT]
+====
+Since this claim was created and bound in the _rh-eng_ project, it can be shared by any user in the project.
+====
+
+Lastly, on GCE console, the new disk will be created and ready for a _Pod_ to be created referencing the persistent volume claim to start using the volume.
+
+
+====
+----
+
+
+kubernetes-dynamic-pvc-e9b4fef7-8bf7-11e6-9962-42010af00004 	SSD persistent disk 	10 GB 	us-east1-d 		
+
+----
+====
+
+
+
+
+[[example2]]
+== Scenario 2: How to enable Default StorageClass behavior for a Cluster
+
+In this example, a cluster-admin will enable a _default_ storage class to be used by all other users and projects that do not implicitly specify a StorageClass annotation in their claim.  This will be useful for a cluster-admin or storage-admin that wants to provide easy management of a storage volume without having to setup or communicate specialized StorageClasses.
+
+
+Building on the previous example, the cluster-admin or storage-admin will create another StorageClass to be designated as the _default_ StorageClass.
+
+.Default StorageClass Spec Definition
+====
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: default <1>
```

Comment:
I don't think I love the name `default` as it could lead someone to believe that is an important/meaninful part of the process.